### DISCUSSION TITLE
ArgoCD Dex Auth with Keycloak Question

### DISCUSSION THREAD

**Discussion by Jonas18175** [argoproj/argo-cd#6596] [General]
Hello,

i setup argocd with keycloak (https://argoproj.github.io/argo-cd/operator-manual/user-management/keycloak/) 
Every is working fine and i can login. I added my user to the ArgoCDAdmins group.

To see all things I setup the deployment to that policy.csv: 
```csv
    p, role:admin, applications, *, */*, allow
    p, role:admin, clusters, *, *, allow
    p, role:admin, repositories, *, *, allow

    g, ArgoCDAdmins, role:admin
    g, ArgoCDReadonly: role:readonly

    # policy.csv is an file containing user-defined RBAC policies and role definitions (optional).
    # Policy rules are in the form:
    #   p, subject, resource, action, object, effect
    # Role definitions and bindings are in the form:
    #   g, subject, inherited-subject
    # See https://github.com/argoproj/argo-cd/blob/master/docs/operator-manual/rbac.md for additional information.
    # policy.csv: |
    #   # Grant all members of the group 'my-org:team-alpha; the ability to sync apps in 'my-project'
    #   p, my-org:team-alpha, applications, sync, my-project/*, allow
    #   # Grant all members of 'my-org:team-beta' admins
    #   g, my-org:team-beta, role:admin
    # policy.default is the name of the default role which Argo CD will falls back to, when
    # authorizing API requests (optional). If omitted or empty, users may be still be able to login,
    # but will see no apps, projects, etc...
    # policy.default: role:readonly
    # scopes controls which OIDC scopes to examine during rbac enforcement (in addition to `sub` scope).
    # If omitted, defaults to: '[groups]'. The scope value can be a string, or a list of strings.
    # scopes: '[cognito:groups, email]'
```

ArgoCD is deployed 'in-cluster', but when I login with my user I see no clusters / projects.
How can I assign the admin group the same permissions like the default admin user? 
Why it doesn't show any in-cluster projects?

**Comment by wtam2018** [comment]
`role:admin` is a built-in role.  You should be able to use it as is.
https://github.com/argoproj/argo-cd/blob/master/docs/operator-manual/rbac.md#basic-built-in-roles
https://github.com/argoproj/argo-cd/blob/master/assets/builtin-policy.csv

**Reply by Jonas18175** [reply]
Yes - but how can I add the in-cluster for the account?
For the local admin account i had to login with argocd-cli and argocd cluster add for that user - but how it works with DEX authed user?

**Comment by Jonas18175** [comment]
I found it out - i made a misstake with the indent of policy.csv in helm values file.

And for the login i had setup a second keycloak client config and added cliClientID: argocd-cli in the oidc.config definition.
